lib/src/emoji_picker_extended_utils.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:emoji_picker_flutter/emoji_picker_flutter.dart';
+import 'package:emoji_picker_flutter/src/recent_emoji.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import 'emoji_picker_internal_utils.dart';
+
+/// Helper class that provides extended usage
+
+class EmojiPickerExtendedUtils {
+  static final List<Emoji> _allAvailableEmojiEntities = [];
+
+  /// Returns list of recently used emoji from cache
+  static Future<List<RecentEmoji>> getRecentEmojis() async {
+    final prefs = await SharedPreferences.getInstance();
+    var emojiJson = prefs.getString('recent');
+    if (emojiJson == null) {
+      return [];
+    }
+    var json = jsonDecode(emojiJson) as List<dynamic>;
+    return json.map<RecentEmoji>(RecentEmoji.fromJson).toList();
+  }
+
+  /// Search for related emoticons based on keywords
+  static Future<List<Emoji>> searchEmoji(String keyword) async {
+    if (keyword.isEmpty) return [];
+
+    if (_allAvailableEmojiEntities.isEmpty) {
+      final emojiPickerInternalUtils = EmojiPickerInternalUtils();
+
+      final availableCategoryEmoji =
+          await emojiPickerInternalUtils.getAvailableCategoryEmoji();
+
+      // Set all the emoji entities
+      availableCategoryEmoji.forEach((_, emojis) {
+        final emojiEntities =
+            emojis.entries.map((emoji) => Emoji(emoji.key, emoji.value));
+        _allAvailableEmojiEntities.addAll(emojiEntities);
+      });
+    }
+
+    return _allAvailableEmojiEntities
+        .where((emoji) => emoji.name.contains(keyword))
+        .toList();
+  }
+
+  /// Add an emoji to recently used list or increase its counter
+  static Future<List<RecentEmoji>> addEmojiToRecentlyUsed(
+      {required Emoji emoji, int recentsLimit = 1000}) async {
```

Comment:
Instead of a random number here I am wondering if it might be better to use `Config` here as parameter. This would make sure `RecentLimit` is same as Config default value and if user wants to have custom `recentLimit` they can just hand over their custom config (same which they use inside the widget, just put it into a separate variable).  What do you think?
```suggestion
      {required Emoji emoji, Config config = const Config()}) async {
```